src/value/mod.rs

```diff
@@ -503,4 +529,7 @@ pub trait Builder {
 
     /// Build a `sexp` from Sequence using Builder
     fn new_sexp(seq: Self::Sequence) -> Self::Element;
+
+    /// Build a `struct` from Struct using Builder
+    fn new_struct(struct_: Self::Struct) -> Self::Element;
```

Comment:
I would change these to just spell it out (see also the implementations).

```suggestion
    fn new_struct(structure: Self::Struct) -> Self::Element;
```